plugins/plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/GitCheckoutHandler.java

```diff
@@ -76,5 +85,8 @@ public void apply(String endpointId, GitCheckoutEventDto dto) {
                 break;
             }
         }
+
+        //Update project attributes from server.
+        resourceManagerFactory.newResourceManager(appContext.getDevMachine()).getWorkspaceProjects();
```

Comment:
Use `appContext.getWorkspaceRoot().synchronize();` instead of this construction.

`org.eclipse.che.ide.resources.impl.ResourceManager.ResourceManagerFactory` it's an internal factory to create a new instance of Resource Manager which not update existing state.